packages/ui-extensions-sdk/src/utils/security.test.ts

```diff
@@ -0,0 +1,127 @@
+import { collectResourceUsage } from './security';
+
+const now = Date.now();
+
+const performance = {
+    getEntriesByType: jest.fn(),
+    timeOrigin: now
+};
+
+Object.defineProperty(window, 'performance', {
+    configurable: true,
+    enumerable: true,
+    value: performance,
+    writable: true
+});
+
+const fakeResourceTimingValues = {
+    connectStart: now + 1,
+    connectEnd: now + 1,
+    domainLookupStart: now + 1,
+    domainLookupEnd: now + 1,
+    fetchStart: now + 1,
+    redirectEnd: 0,
+    redirectStart: 0,
+    requestStart: now + 1,
+    responseStart: now + 2,
+    responseEnd: now + 30,
+    secureConnectionStart: 0,
+    decodedBodySize: 0,
+    encodedBodySize: 0,
+    nextHopProtocol: 'http',
+    serverTiming: [],
+    transferSize: 0,
+    workerStart: 0,
+    toJSON() {
+        throw new Error('Function not implemented.');
+    },
+    duration: 0,
+    startTime: 0,
+    entryType: 'resource'
+};
+
+const fakeImageResource: PerformanceResourceTiming = {
+    ...fakeResourceTimingValues,
+    ...{
+        initiatorType: 'image',
+        name: 'http://localhost/image/1'
+    }
+};
+
+const fakeRequestResource: PerformanceResourceTiming = {
+    ...fakeResourceTimingValues,
+    ...{
+        initiatorType: 'fetch',
+        name: 'http://localhost/api/1'
+    }
+};
+
+beforeEach(() => {
+    performance.getEntriesByType.mockReset();
+});
+
+describe('collectResourceUsage', () => {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏These tests are great!